### ISSUE TITLE
Support for pages

### ISSUE THREAD

**Issue by dziudek** [GetPublii/Publii#656]
We will separate posts from pages for better content management.

**Comment by firedev** [comment]
Could you please share some ideas how you plan to approach this? Separate `pages` table I guess? Then what with the meta data – add more `pages_*` set of tables similar to `posts_*` tables? Or start with `page_id` column in `posts_` tables at first and then refactor to the join table?

**Comment by tiotrom** [comment]
I didn't see the feature in the latest release. Do you plan to do it for the next release? Cheers!

**Comment by cayblood** [comment]
This is the main missing feature for me to be able to use this for my blog.

**Comment by bobmitro** [comment]
> This is the main missing feature for me to be able to use this for my blog.

Just mark the post as Hidden, then it will work as a page; additionally, you can change the HTML structure of this post by creating a custom post-template.

**Comment by cayblood** [comment]
Thanks for your response, @bobmitro. That is a reasonable workaround, but it would be much nicer to simply have an interface to create a custom page with full html editing for any URI.

**Comment by cayblood** [comment]
@dziudek you marked this feature as belonging to the 1.0.0 milestone, but there doesn't seem to be an actual milestone for this version number. Do you have any ballpark idea of when 1.0.0 will be tagged? What major features are still missing?

**Comment by matthillco** [comment]
I'm a new Publii user, I love it so far as a blogging system. For more sophisticated websites, I agree that it does need different content types like pages and nested pages that aren't part of the "blog" content. I know that pages aren't on the roadmap right now, but I wonder if there's any news on whether they might be added in the future?

**Comment by dziudek** [comment]
We are exploring the topic of custom content types (including Pages). We want to implement them in one of the next releases, but we will do it gradually, so for 99% without nesting in the first phase. But yes - we see that Publii needs a way to separate content in a different way than tags :)

**Comment by matthillco** [comment]
That's sounds great @dziudek! I'm not a back-end dev, I'm mostly a front-end designer (HTML+CSS), but is there anything I could do to help with this?

**Comment by egrieco** [comment]
I got really excited when I first saw Publii and was pretty impressed upon trying it out (nice editor). However, I'm not currently using it or recommending it much since without the ability to do basic page editing and hierarchies, it's not really a workable solution for most blogs or lightweight sites.

If decent hierarchical page support is added, I'll be singing its praises from the rooftops.

**Comment by jotamorais** [comment]
@dziudek I [posted in the Publii's forum](https://forum.getpublii.com/topic/how-to-create-custom-objects-beyond-the-built-ins-featuredimage-author-maintag/#post-8026) minutes ago asking about something like this.

Is this still a plan? Anything the community could help with to get it off the ground?

**Comment by henkvantijen** [comment]
I miss the 'pages' feature also. Funny that many of the real-life sample sites are that: static sites, not blogs or news like sites.
Maybe a workaround is a possible solution:
When creating a new site, a Dialog appears , asking _Do you want to create (A) a blog/news site or (B) a site with static content._ When user choses (B), a number of settings are set :
- posts will be  hidden by default
- no previous next articles , or related articles,
- nr of posts : 0
- a menu is created, with 1 entry, home
- a editable home page is created , linked to the main menu item
- no date based content list
(above may contain errors, I am not too familiar to Publii)

And, also, a little HOWTO tutorial or FAQ page could be dedicated  to the creation of a non-posts site. This is the quickest way to help users.

**Comment by bobmitro** [comment]
As I wrote above, mark the post as Hidden, then it will work as a page; additionally, you can change the post HTML structure by creating a custom post-template.

**Comment by egrieco** [comment]
> As I wrote above, mark the post as Hidden, then it will work as a page; additionally, you can change the post HTML structure by creating a custom post-template.

This is a decent-ish workaround when you have a very small number of actual pages, but it falls apart when you have a site with even moderate hierarchy e.g. a docs website or personal wiki.

At this point, I'm investigating writing my own Publii replacement that works with flat files (Git Repos) and natively supports hierarchy. Not a bad project to kick the tires on a few Rust GUI libraries.

**Comment by dziudek** [comment]
@egrieco - or you can wait 2-3 weeks as https://github.com/GetPublii/Publii/tree/feature-pages will be merged into v.0.46 ;)

**Comment by egrieco** [comment]
> ...wait 2-3 weeks...

Yeah, I doubt I'll get to implementation in that time frame. Will definitely take a look when it drops.

Now, if the backing store was flat files instead of a sqlite database we'd be totally good to go. 😄

**Comment by egrieco** [comment]
Any word on the v.0.46 release? Curious to see how it was implemented.

P.S. Totally understand delays, I'm behind on a few projects at the moment myself.

**Comment by bobmitro** [comment]
v0.46 is coming; it should be ready around August 9-11. There is a lot to do, and there are only two of us. All themes are being rebuilt, plugins are being updated, documentation is being updated, etc.

Here is a sneak peek of the updated [Simple v.3](https://demo.getpublii.eu/themes/simple/v3/) theme, showcasing the latest Publii v0.46 features, including Pages.

**Comment by egrieco** [comment]
I see that `v0.46` just dropped. Downloading it now. 😄

**Comment by egrieco** [comment]
Haven't done extensive testing of the pages feature yet, mostly just getting acquainted with Publii right now.

So far, the pages feature is working well though. 🎉

**Comment by bobmitro** [comment]
Closed: This is now part of Publii [v. 0.46](https://getpublii.com/blog/release-046.html)
